flowzone.yml

```
@@ -1547,31 +1549,19 @@ jobs:
         shell: bash --noprofile --norc -eo pipefail -x {0}
 
     steps:
-      # https://github.com/dawidd6/action-download-artifact
-      # TODO: what if this is a tag event and PR artifacts do not exist?
-      # We want to use the source artifact from the PR instead of the merge
-      # action so the release notes don't include the merge commit
-      - name: Download source artifact from PR
-        uses: dawidd6/action-download-artifact@v2
-        with:
-          github_token: ${{ secrets.FLOWZONE_TOKEN }}
-          commit: ${{ github.event.pull_request.head.sha || github.event.head_commit.id }}
-          path: ${{ runner.temp }}
-          workflow_conclusion: success
-          name: source-${{ github.event.pull_request.head.sha || github.event.head_commit.id }}
-
+      - *downloadSourceArtifact
       - *extractSourceArtifact
 
       # https://docs.github.com/en/rest/releases
       - name: Finalize GitHub release (if any)
         if: needs.versioned_source.outputs.version != ''
-        shell: bash
         run: |
           set -ea
           # prevent git from existing with 141
           set +o pipefail
           previous_tag="$(git tag --sort=-version:refname | head -n 2 | tail -n 1)"
-          release_notes="$(git log ${previous_tag}..HEAD --pretty=reference)"
+          # Use HEAD~2 to skip the merge and versioning commit
+          release_notes="$(git log ${previous_tag}..HEAD~2 --pretty=reference)"
```

This isn't bad, but would it be even better to use the head commit id instead of `HEAD~2`?
```suggestion
          release_notes="$(git log ${previous_tag}..${{ github.event.pull_request.head.sha || github.event.head_commit.id }} --pretty=reference)"
```
Maybe you tried this already, but I think the above would always be the last commit from the PR?

Otherwise `HEAD~2` is fine and unlikely to cause problems.

---

That's smart, I didn't think of that. I'll make the change